syllabus/core/curriculum_base.py

```diff
@@ -151,23 +151,22 @@ def _sample_distribution(self) -> List[float]:
         raise NotImplementedError
 
     def _should_use_startup_sampling(self) -> bool:
-        return self.use_startup_behavior 
+        return self.warmup_strategy != "none" and self.sampled_tasks < self.warmup_tasks
     
     def _startup_sample(self, k: int) -> List:
         sampled_tasks = []
-        if self.sampled_strategy == "fix":
-            end_index = min(self.current_task_index + k, len(self.tasks))
-            sampled_tasks.extend(self.tasks[self.current_task_index:end_index])
-            self.current_task_index = end_index
+        if self.warmup_strategy == "fix":
+            end_index = min(self.sampled_tasks + k, self.num_tasks)
+            sampled_tasks = self.tasks[self.sampled_tasks:end_index]
+            self.sampled_tasks = end_index
 
-        if self.sampled_strategy == "random":
-            num_tasks = len(self.tasks)
+        if self.warmup_strategy == "random":
+            num_tasks = self.num_tasks
             if k > num_tasks:
                 raise ValueError("k cannot be larger than the total number of tasks")
```

Comment:
k can be greater than num_tasks here, we'll just sample some tasks more than once